assets/js/phoenix_live_view.js

```diff
@@ -313,19 +313,19 @@ export class LiveSocket {
     })
   }
 
-  replaceRoot(href, linkRef = this.setPendingLink(href)){
+  replaceRoot(href, callback = null, linkRef = this.setPendingLink(href)){
```

Comment:
Now that `callback` is optional, this code will no longer call `Browser.push_state`:

https://github.com/phoenixframework/phoenix_live_view/blob/b0d760c42a090d0920a2b054c9fd736c13e2df7a/assets/js/phoenix_live_view.js#L489

My first hunch is that the new code is correct. After all, we don't want to push something when the state was just popped? But I have no idea how this works. :)

Reply:
If the user has used forward/back to trigger pop state, we need to push the event to the server to be invoked by handle_params (or to tell is to go external and replace root)

Reply:
In the code above we still do so. The difference is that prior to this patch, we would go to the server AND call pushState. Now we only go to the server.